src/rgw/rgw_op.cc

```diff
@@ -4537,8 +4537,16 @@ void RGWPutACLs::execute()
   owner = existing_policy->get_owner();
 
   op_ret = get_params();
-  if (op_ret < 0)
+  if (op_ret < 0) {
+    if (op_ret == -ERANGE) {
+      ldout(s->cct, 0) << "RGWPutACLs::" << __func__
+                       << " The size of request xml data is beyond over the max len limit, data size = "
+                       << atoll(s->length) << dendl;
+      op_ret = -ERR_MALFORMED_XML;
+      s->err.message = "The XML you provided was not well-formed or did not validate against our published schema";
```

Comment:
is it useful to return the exact same string as aws here? i would prefer something more specific, like `The XML you provided was larger than the maximum <X> bytes allowed`